engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/KvmNonManagedStorageDataMotionStrategy.java

```diff
@@ -140,4 +160,56 @@ protected void setVolumePath(VolumeVO volume) {
     protected boolean shouldMigrateVolume(StoragePoolVO sourceStoragePool, Host destHost, StoragePoolVO destStoragePool) {
         return sourceStoragePool.getPoolType() == StoragePoolType.Filesystem;
     }
+
+    /**
+     * If the template is not on the target primary storage then it migrates the template.
+     * Otherwise the execution flow fails during migration and creates a null file created on the target storage pool.
```

Comment:
this comments confuses me. you wrote in the description that the template would be copied and this problem would no longer be there. if that is true this "Otherwise ..." is not true or badly formulated. Can you explain?
As it is written now it seems that the condition after "Otherwise" can still occur.
If I understand this correctly you should start with "If we don't do this ...", or "This is needed to prevent ...". Am I correct?

Reply:
You are right @DaanHoogland. I will remove the `Otherwise the execution flow fails during migration and creates a null file created on the target storage pool.` from the Javadoc and leave just the `If the template is not on the target primary storage then it migrates the template.`. Thanks for pointing it!

Reply:
Done!